mirascope/chat/models.py

```diff
@@ -90,3 +103,40 @@ def stream(
 
         for chunk in completion_stream:
             yield OpenAIChatCompletionChunk(chunk=chunk)
+
+    def extract(
+        self, prompt: Union[Prompt, str], schema: Type[BaseModelT], retries: int = 0
+    ) -> BaseModelT:
+        """Extracts the given schema from the response of a chat `create` call.
+
+        Args:
+            prompt: The `Prompt` from which the schema will be extracted.
+            schema: The `BaseModel` schema to extract from the completion.
+            retries: The maximum number of times to retry the query on validation error.
+
+        Returns:
+            The `Schema` instance extracted from the completion.
+
+        Raises:
+            ValidationError: if the schema cannot be instantiated from the completion.
+        """
+        tool = convert_base_model_to_openai_tool(schema)
+        completion = self.create(
+            prompt,
+            tools=[tool],
+            tool_choice={
+                "type": "function",
+                "function": {"name": tool.__name__},
+            },
+        )
+
+        try:
```

Comment:
BONUS: consider extracting (pun intended) these to a helper function so its easier to implement for async extract

Reply:
I'll punt the refactor here to when implementing async extract